pkg/source/chirpstack/source.go

```diff
@@ -205,30 +192,23 @@ func (p *Source) ExportDevice(devEui string) (*ttnpb.EndDevice, error) {
 	}
 
 	// Join (OTAA/ABP)
-	dev.SupportsJoin = devProfile.SupportsJoin
+	dev.SupportsJoin = devProfile.SupportsOtaa
 	if !dev.SupportsJoin {
-		if devProfile.RxFreq_2 > 0 {
+		if freq := devProfile.AbpRx2Freq; freq > 0 {
 			dev.MacSettings.Rx2Frequency = &ttnpb.FrequencyValue{
-				Value: uint64(devProfile.RxFreq_2),
+				Value: uint64(freq),
 			}
 		}
-		if devProfile.RxDelay_1 > 0 {
+		if delay := devProfile.AbpRx1Delay; delay > 0 {
 			dev.MacSettings.Rx1Delay = &ttnpb.RxDelayValue{
-				Value: ttnpb.RxDelay(devProfile.RxDelay_1),
+				Value: ttnpb.RxDelay(delay),
 			}
 		}
-		if devProfile.RxDrOffset_1 >= 0 {
-			dev.MacSettings.DesiredRx1DataRateOffset = &ttnpb.DataRateOffsetValue{
-				Value: ttnpb.DataRateOffset(devProfile.RxDrOffset_1),
-			}
+		dev.MacSettings.DesiredRx1DataRateOffset = &ttnpb.DataRateOffsetValue{
```

Comment:
I would set this only if it is greater than zero as it was originally done. Also are we sure it is desired ?

Reply:
`DesiredRx1DataRateOffset ` is unsigned now, so I removed the `>=` condition. Should those be kept for the sake of verbosity?

> is it a desired parameter

It's hard for me to judge that, I didn't want to remove the features we had here previously and compared the values from CS v3 to v4 migration script.

Reply:
I would set this only if it is greater than 0, and I believe that the original implementation was wrong, and this is actually `Rx1DataRateOffset`, not `DesiredRx1DataRateOffset`.